ingestion/src/metadata/utils/ssl_manager.py

```diff
@@ -147,7 +170,22 @@ def _(self, connection):
 
 
 @singledispatch
-def check_ssl_and_init(_):
+def check_ssl_and_init(_) -> None:
+    return None
```

Comment:
I suggest to `raise NotImplementedError` for the base case

Reply:
I was thinking of doing a whole cleanup, so that we need to initialize and modify things at single place
It will take place as part of this #15913